pkg/util/log/testdata/log_migration

```diff
@@ -0,0 +1,50 @@
+structured-event should_migrate=true new_channel=OPS
+----
+DEV Channel: 
+[]
+OPS Channel: 
+[{Severity:INFO Time:0 Goroutine:0 File:util/log_test/migrator_test.go Line:74 Message:Structured entry: {"Timestamp":1735689600000000000,"EventType":"test_event"} Tags: Counter:2 Redactable:true Channel:OPS StructuredEnd:76 StructuredStart:18 StackTraceStart:0 TenantID: TenantName:}]
+
+structured-event should_migrate=false new_channel=OPS
+----
+DEV Channel: 
+[{Severity:INFO Time:0 Goroutine:0 File:util/log_test/migrator_test.go Line:74 Message:Structured entry: {"Timestamp":1735689600000000000,"EventType":"test_event"} Tags: Counter:5 Redactable:true Channel:DEV StructuredEnd:76 StructuredStart:18 StackTraceStart:0 TenantID: TenantName:}]
+OPS Channel: 
+[]
+
+structured-event should_migrate=true new_channel=OPS depth=0
+----
+DEV Channel: 
+[]
+OPS Channel: 
+[{Severity:INFO Time:0 Goroutine:0 File:util/log_test/migrator_test.go Line:72 Message:Structured entry: {"Timestamp":1735689600000000000,"EventType":"test_event"} Tags: Counter:8 Redactable:true Channel:OPS StructuredEnd:76 StructuredStart:18 StackTraceStart:0 TenantID: TenantName:}]
+
+structured-event should_migrate=false new_channel=OPS depth=0
+----
+DEV Channel: 
+[{Severity:INFO Time:0 Goroutine:0 File:util/log_test/migrator_test.go Line:72 Message:Structured entry: {"Timestamp":1735689600000000000,"EventType":"test_event"} Tags: Counter:11 Redactable:true Channel:DEV StructuredEnd:76 StructuredStart:18 StackTraceStart:0 TenantID: TenantName:}]
+OPS Channel: 
+[]
+
+structured-event should_migrate=true new_channel=OPS depth=1
+----
+DEV Channel: 
+[]
+OPS Channel: 
+[{Severity:INFO Time:0 Goroutine:0 File:datadriven/datadriven.go Line:119 Message:Structured entry: {"Timestamp":1735689600000000000,"EventType":"test_event"} Tags: Counter:14 Redactable:true Channel:OPS StructuredEnd:76 StructuredStart:18 StackTraceStart:0 TenantID: TenantName:}]
```

Comment:
It's not great to have tests that depend on line numbers in general libraries (especially when they are external dependencies). Can we "censor" the line number at least?

Reply:
@RaduBerinde Yeah i agree, here is a PR to address this https://github.com/cockroachdb/cockroach/pull/154874